backend/app/api/routes/graph.py

```
@@ -47,16 +52,25 @@ async def _get_evaluation(evaluation_id: str) -> dict[str, Any] | None:
 
     Raises:
-        CorkedError: If user does not own the evaluation.
+        CorkedError: If user does not own the evaluation and it's not public.
     """
+    # Allow access to public demo evaluations without auth
+    if evaluation_id in PUBLIC_DEMO_EVALUATIONS:
+        return
+    
+    # Require auth for non-public evaluations
+    if user is None:
+        raise CorkedError("Authentication required to view this evaluation")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)